### PR TITLE
Feature/zms 288

### PR DESCRIPTION
**Pull request by grishick**
ZContact will stream VCARD from mailstore to IMAP similar to ZMessage. ZMessage uses  ContentServlet, which does not support Contacts, but UserServlet does, so using UserServlet instead of ContentServlet.

### CODE REVIEW

File: store/src/java/com/zimbra/cs/service/mail/GetContacts.java [hunk 1]
```diff
@@ -74,48 +79,50 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
...
+        if(reqAttrs != null && reqAttrs.size() > 0) {
+            memberAttrs = new ArrayList<String>();
+            for(AttributeName attrName : reqAttrs) {
+                memberAttrs.add(attrName.getName());
+            }
+        }
+
+        //MailConstants.E_CONTACT
+        List<Id> contactIds = req.getContacts();
+        if(contactIds != null && contactIds.size() > 0) {
```
Comment (fatkudu) [inline store/src/java/com/zimbra/cs/service/mail/GetContacts.java]: It looks to me as if there is a bug in the JAXB.  The old code appears to allow multiple E_CONTACT elements, each with an id attribute which may be a comma separated list of IDs.  That seems a bit weird but we should probably preserve that behavior?  So, `req.getContacts()` should return `List<IdList>` where `IdList` is a new type whose ID attribute can be a list of Ids?

File: store/src/java/com/zimbra/cs/service/mail/GetContacts.java [hunk 2]
```diff
@@ -203,7 +210,14 @@ static void partitionItems(ZimbraSoapContext lc, ArrayList<ItemId> ids, ArrayLis
     throws ServiceException {
         List<Element> responses = new ArrayList<Element>();
 
-        // note that we removed all <cn> elements from the request during handle(), above
+        //remove all 'contact' elements from original request
+        for (Element e : request.listElements()) {
```
Comment (fatkudu) [inline store/src/java/com/zimbra/cs/service/mail/GetContacts.java]: There is a `List<Element> listElements(String name)` method too which may be worth using?

Comment (grishick) [inline store/src/java/com/zimbra/cs/service/mail/GetContacts.java]: old code was also detaching elements whilst looping (line 107 in old code), so I assumed it worked :)

File: store/src/java/com/zimbra/cs/service/mail/GetContacts.java (hunk 1 above)
Comment (grishick) [inline store/src/java/com/zimbra/cs/service/mail/GetContacts.java]: you are right. soap.txt:

<GetContactsRequest [sortBy="nameAsc|nameDesc"] [sync="1"] [l="{folder-id}"] [derefGroupMember="1"] 
    [returnHiddenAttrs="1"] [maxMembers="{max-members}"] [returnCertInfo="1"]>
  <a n="..."/>*
  <ma n="..."/>*
  <cn id="{contact-id}[,{contact-id}...]"/>*
</GetContactsRequest>

**Pull request closed without merging**